server/graphql/invoice/src/mutations/prescription/insert.rs

```diff
@@ -0,0 +1,307 @@
+use async_graphql::*;
+
+use graphql_core::simple_generic_errors::{OtherPartyNotAPatient, OtherPartyNotVisible};
+use graphql_core::standard_graphql_error::{validate_auth, StandardGraphqlError};
+use graphql_core::ContextExt;
+use graphql_types::types::InvoiceNode;
+use repository::Invoice;
+use service::auth::{Resource, ResourceAccessRequest};
+use service::invoice::prescription::{
+    InsertPrescription as ServiceInput, InsertPrescriptionError as ServiceError,
+};
+
+#[derive(InputObject)]
+#[graphql(name = "InsertPrescriptionInput")]
+pub struct InsertInput {
+    pub id: String,
+    pub patient_id: String,
+}
+
+#[derive(SimpleObject)]
+#[graphql(name = "InsertPrescriptionError")]
+pub struct InsertError {
+    pub error: InsertErrorInterface,
+}
+
+#[derive(Union)]
+#[graphql(name = "InsertPrescriptionResponse")]
+pub enum InsertResponse {
+    Error(InsertError),
+    Response(InvoiceNode),
+}
+
+pub fn insert(ctx: &Context<'_>, store_id: &str, input: InsertInput) -> Result<InsertResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateOutboundShipment,
```

Comment:
OG uses this permission for prescriptions, so thought to re-use instead of mapping out a new permission for prescriptions

Reply:
I would actually define a new Resource::MutatePrescription and map this with to the OG error

Reply:
[afa1d99](https://github.com/openmsupply/open-msupply/pull/1965/commits/afa1d9940e1391777cb415c870cb9a1d24327f9c)